### PR TITLE
feat(ST-146): implement assignment submissions endpoint

### PR DESCRIPTION
**Pull request by ghost**
## Description of Change

- Created AssignmentSubmission controller, service and module
- Added endpoint for submitting assignment
- Created Dto file for assignment submission
- Updated repository to upload on the database

## Associated Ticket Link(s)
- https://trello.com/c/LSF9sz2N

## Screenshots / Screen Recordings
N/A

### CODE REVIEW

File: src/assignment_submissions/assignment_submissions.controller.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import { AssignmentSubmissionsService } from "./assignment_submissions.service";
+
+@Controller("classrooms")
+export class AssignmentSubmissionsController {
+  constructor(
+    private readonly assignmentSubmissionsService: AssignmentSubmissionsService,
+    private readonly assignmentSubmissionsSerializer: AssignmentSubmissionsSerializer,
+  ) {}
+
+  @Post("submit-assignment")
```
Comment (ishmam-iqbal-sazim) [inline src/assignment_submissions/assignment_submissions.controller.ts:15]: Let's follow REST principles


Endpoint should be something like `classrooms/:classroomId/assignments/:assignmentId/submit`

Comment (ishmam-iqbal-sazim) [inline src/assignment_submissions/assignment_submissions.controller.ts:15]: Shouldn't only students be able to submit assignments?

File: src/assignment_submissions/assignment_submissions.dtos.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+import { IsNumber } from "class-validator";
+
+export class CreateAssignmentSubmissionDto {
+  @IsNumber()
+  @Type(() => Number)
+  assignmentId!: number;
+
+  @IsNumber()
+  @Type(() => Number)
+  userId!: number;
```
Comment (ishmam-iqbal-sazim) [inline src/assignment_submissions/assignment_submissions.dtos.ts:11]: Do you need this?
Can't you use the Current user decorator?

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Approved pending comments

File: src/assignment_submissions/assignment_submissions.dtos.ts (hunk 2 above)
Comment (ghost) [reply src/assignment_submissions/assignment_submissions.dtos.ts:11]: Yes, definitely.  At first I was taking studentID, but then I have changed to userID.

File: src/assignment_submissions/assignment_submissions.controller.ts (hunk 1 above)
Comment (ghost) [reply src/assignment_submissions/assignment_submissions.controller.ts:15]: Yes, I will use RolesGuard

**Pull request merged**